src/cuda/tile/_passes/token_order.py

```diff
@@ -150,15 +150,16 @@ def get_memory_effects(cur_op):
 
         if isinstance(cur_op, LoadMemoryOperation):
             effect = MemoryEffect.LOAD
-        else:
-            assert isinstance(cur_op, StoreMemoryOperation)
+        elif isinstance(cur_op, StoreMemoryOperation):
             effect = MemoryEffect.STORE
+        else:
+            raise TileInternalError(f"Unexpected MemoryOperation type: {type(cur_op)}")
 
         has_acquire_order = False
         if isinstance(cur_op, (TileAtomicCAS, TileAtomicRMW)):
-            has_acquire_order = memory_order_has_acquire(op.memory_order)
+            has_acquire_order = memory_order_has_acquire(cur_op.memory_order)
```

Comment:
Should this be the outer or inner variable? I tried to read through the code and could not find any cases where we did not want the `cur_op`